lib/grpc-eio/connection.ml

```diff
@@ -0,0 +1,50 @@
+let grpc_recv_streaming body message_buffer_writer =
+  let request_buffer = Grpc.Buffer.v () in
+  let on_eof () = Seq.close_writer message_buffer_writer in
+  let rec on_read buffer ~off ~len =
+    Grpc.Buffer.copy_from_bigstringaf ~src_off:off ~src:buffer
+      ~dst:request_buffer ~length:len;
+    let message = Grpc.Message.extract request_buffer in
+    let () =
+      match message with
+      | Some message -> Seq.write message_buffer_writer message
+      | None -> ()
+    in
+    H2.Body.Reader.schedule_read body ~on_read ~on_eof
+  in
+  H2.Body.Reader.schedule_read body ~on_read ~on_eof
+
+let grpc_send_streaming_client body encoder_stream =
+  Seq.iter
+    (fun encoder ->
+      let payload = Grpc.Message.make encoder in
+      H2.Body.Writer.write_string body payload)
```

Comment:
That's a good question. It could be preferable when aiming for low latency. I'm thinking maybe it should just be opt-in.

On this note, we ran some load tests but they were inconclusive because we stumbled upon a bug in Eio: https://github.com/ocaml-multicore/eio/issues/572
It has been fixed, so I hope we'll have better numbers soon.

We're also experimenting with a different interface for the reader. Lazy sequence might be a bit too opinionated, and they introduce a bit of overhead, so the alternative interface is similar to H2 with `on_msg`/`on_eof` callbacks, with a conversion to `Seq.t` as a convenience helper: https://github.com/dialohq/ocaml-grpc/tree/grpc-eio-noseq/lib/grpc-eio

Reply:
Interesting experiment, it will be good to see some performance numbers from it. Keeping the `Seq.t` for a simple interface is a good idea for introducing the library and less performance sensitive applications. 

Opt-in sounds good, I'm happy to leave that till later on. There are a few other config pieces like auth and compression that will need a way to configure them.